src/lib/functions/netlify-function.js

```diff
@@ -114,7 +114,12 @@ class NetlifyFunction {
   }
 
   get url() {
-    return `http://localhost:${this.settings.port}/.netlify/functions/${this.name}`
+    // This line fixes the issue here https://github.com/netlify/cli/issues/4116
+    // Not sure why `settings.port` was used here nor does a valid reference exist.
+    // However, it remains here to serve whatever purpose for which it was added.
+    const port = this.settings.port || this.settings.functionsPort
+
+    return `http://localhost:${port}/.netlify/functions/${this.name}`
```

Comment:
Maybe you want to construct the url based on the `URL`.

```
new URL({
  port,
  ...
}).href
```

Reply:
@erezrokah are the functions always served on the host `localhost` or could it be `0.0.0.0` as well?